Source/gmenu.cpp

```diff
@@ -126,38 +126,35 @@ void gmenu_call_proc(TMenuItem *pItem, void(*gmFunc)(TMenuItem *))
 		}
 	}
 	sgpCurrItem = &v2[v3 - 1];
-	gmenu_up_down(1);
+	gmenu_up_down(TRUE);
 }
 // 525740: using guessed type int PauseMode;
 // 634464: using guessed type char byte_634464;
 // 63448C: using guessed type int dword_63448C;
 
-void gmenu_up_down(int a1)
+void gmenu_up_down(BOOL isDown)
 {
-	TMenuItem *v1; // eax
-	int v2;        // edi
+	int i;
 
-	v1 = sgpCurrItem;
-	if (sgpCurrItem) {
-		byte_634464 = 0;
-		v2 = dword_63448C;
-		while (v2) {
-			--v2;
-			if (a1) {
-				++v1;
-				sgpCurrItem = v1;
-				if (v1->fnMenu)
-					goto LABEL_10;
-				v1 = dword_634480;
+	if (!sgpCurrItem) {
+		return;
+	}
+	byte_634464 = FALSE;
+	i = dword_63448C;
+	if (dword_63448C) {
+		while (i) {
+			i--;
+			if (isDown) {
+				sgpCurrItem++;
+				if (!sgpCurrItem->fnMenu)
+					sgpCurrItem = dword_634480;

```

Comment:
```suggestion
					sgpCurrItem = &dword_634480[0];
```

Reply:
why like this? array's name is actually a reference to the first element :<

Reply:
To make it clear that we want the first menu item and not the entire menu.